gtsam/hybrid/tests/testHybridLookupDAG.cpp

```diff
@@ -0,0 +1,272 @@
+  DiscreteKey X(0, 2), Y(1, 2);
+
+  auto conditional = boost::make_shared<DiscreteConditional>(X | Y = "0/1 3/2");
+
+  HybridConditional hc(conditional);
+
+  HybridLookupTable hlt(hc);
+
+  DiscreteValues dv;
+  dv[1] = 0;
+  VectorValues cv;
+  // cv.insert(X(2),Vector2(0.0, 0.0));
+  HybridValues hv(dv, cv);
+
+  hlt.argmaxInPlace(&hv);
+
+  EXPECT(assert_equal(hv.atDiscrete(0), 1));
+
+  DecisionTreeFactor f1(X, "2 3");
+  auto conditional2 = boost::make_shared<DiscreteConditional>(1, f1);
+
+  HybridConditional hc2(conditional2);
+
+  HybridLookupTable hlt2(hc2);
+
+  HybridValues hv2;
+
+  hlt2.argmaxInPlace(&hv2);
+
+  EXPECT(assert_equal(hv2.atDiscrete(0), 1));
+}
+
+TEST(HybridLookupTable, gaussian_argmax) {
+  Matrix S1(2, 2);
+  S1(0, 0) = 1;
+  S1(1, 0) = 0;
+  S1(0, 1) = 0;
+  S1(1, 1) = 1;
+
+  Vector2 d1(0.2, 0.5), d2(-0.5, 0.6);
+
+  SharedDiagonal model = noiseModel::Diagonal::Sigmas(Vector2(1.0, 0.34));
+
+  auto conditional =
+      boost::make_shared<GaussianConditional>(X(1), d1, S1, X(2), -S1, model);
+
+  HybridConditional hc(conditional);
+
+  HybridLookupTable hlt(hc);
+
+  DiscreteValues dv;
+  // dv[1]=0;
+  VectorValues cv;
+  cv.insert(X(2), d2);
+  HybridValues hv(dv, cv);
+
+  hlt.argmaxInPlace(&hv);
+
+  EXPECT(assert_equal(hv.at(X(1)), d1 + d2));
+}
+
+TEST(HybridLookupDAG, argmax) {
+  Matrix S1(2, 2);
+  S1(0, 0) = 1;
+  S1(1, 0) = 0;
+  S1(0, 1) = 0;
+  S1(1, 1) = 1;
+
+  Vector2 d1(0.2, 0.5), d2(-0.5, 0.6);
+
+  SharedDiagonal model = noiseModel::Diagonal::Sigmas(Vector2(1.0, 0.34));
+
+  auto conditional0 =
+           boost::make_shared<GaussianConditional>(X(2), d1, S1, model),
+       conditional1 =
+           boost::make_shared<GaussianConditional>(X(2), d2, S1, model);
+
+  DiscreteKey m1(1, 2);
+  GaussianMixture::Conditionals conditionals(
+      {m1},
+      vector<GaussianConditional::shared_ptr>{conditional0, conditional1});
+  boost::shared_ptr<GaussianMixture> mixtureFactor(
+      new GaussianMixture({X(2)}, {}, {m1}, conditionals));
+  HybridConditional hc2(mixtureFactor);
+  HybridLookupTable hlt2(hc2);
+
+  auto conditional2 =
+      boost::make_shared<GaussianConditional>(X(1), d1, S1, X(2), -S1, model);
+
+  HybridConditional hc1(conditional2);
+  HybridLookupTable hlt1(hc1);
+
+  DecisionTreeFactor f1(m1, "2 3");
+  auto discrete_conditional = boost::make_shared<DiscreteConditional>(1, f1);
+
+  HybridConditional hc3(discrete_conditional);
+  HybridLookupTable hlt3(hc3);
+
+  HybridLookupDAG dag;
+  dag.push_back(hlt1);
+  dag.push_back(hlt2);
+  dag.push_back(hlt3);
+  auto hv = dag.argmax();
+
+  EXPECT(assert_equal(hv.atDiscrete(1), 1));
+  EXPECT(assert_equal(hv.at(X(2)), d2));
+  EXPECT(assert_equal(hv.at(X(1)), d2 + d1));
+}
+
+/* ************************************************************************* */
+int main() {
+  TestResult tr;
+  return TestRegistry::runAllTests(tr);
+}
+/* ************************************************************************* */
```

Comment:
newline

Reply:
This is from @xsj01's PR. @xsj01 seems like your formatter isn't set up properly yet.